packages-tests/NodeNameResolver/NodeNameResolverTest.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Tests\NodeNameResolver;
+
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use Rector\NodeNameResolver\NodeNameResolver;
+use Rector\Testing\PHPUnit\AbstractTestCase;
+
+final class NodeNameResolverTest extends AbstractTestCase
+{
+    private NodeNameResolver $nodeNameResolver;
+
+    protected function setUp(): void
+    {
+        $this->boot();
+        $this->nodeNameResolver = $this->getService(NodeNameResolver::class);
+    }
+
+    public function testGetNameOnMethodCallWithIdentifier(): void
+    {
+        $methodCallWithNameIdentifier = new MethodCall(
+            new Variable('foo'),
+            new Identifier('bar')
+        );
+        $this->assertSame('bar', $this->nodeNameResolver->getName($methodCallWithNameIdentifier->name));
+        $this->assertSame('bar', $this->nodeNameResolver->getName($methodCallWithNameIdentifier));
```

Comment:
This line should throw exception, as method call is not a nameable node.

E.g. property fetch is not nameable node either:

```php
$this->value
```

What would be a name? `this`? `value` or `this->value`?

There was lot of bugs with assumption, that MethodCall returns left or right part, so this was added to make sure the `name` is needed, nod the variable part. Mainly in chained calls this caused mess:

```php
$this->createBuilder()
    ->select('...')
    ->from(some()->value())`
```

So we better go narrow than open space for bugs :)

Reply:
Exactly, it should throw exception :)